src/Domain/CoordinateMaps/KerrHorizon.hpp

```diff
@@ -0,0 +1,68 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Utilities/ContainerHelpers.hpp"
+#include "Utilities/DereferenceWrapper.hpp"
+#include "Utilities/MakeWithValue.hpp"
+#include "Utilities/TypeTraits/RemoveReferenceWrapper.hpp"
+
+namespace domain::CoordinateMaps {
+
+/*!
+ * Distorts cartesian coordinates \f$X^i\f$ such that a coordinate sphere
+ * \f$\delta_{ij}X^iX^j=C\f$ is mapped to an ellipsoid of constant
+ * Kerr-Schild-radius \f$r\f$, defined as the largest positive root of
+ *
+ *  \f{equation*}{
+ *  r^4 - r^2 (\vec{x}^2 - \vec{a}^2) - (\vec{a}\cdot \vec{x})^2 = 0.
+ *  \f}
+ *
+ * In this equation, \f$\vec{x}\f$ is the coordinate vector to the (distorted)
```

Comment:
[picking line at random] could you document what the actual math is that the coordinate map is doing? that is, what are `target_coords` as a function of `source_coords`? Please also add what the (inv) jacobian is